core/src/main/java/com/datastrato/gravitino/catalog/SchemaOperationDispatcher.java

```
@@ -297,20 +298,35 @@ public Schema alterSchema(NameIdentifier ident, SchemaChange... changes)
    */
   @Override
   public boolean dropSchema(NameIdentifier ident, boolean cascade) throws NonEmptySchemaException {
-    boolean dropped =
+    NameIdentifier catalogIdent = getCatalogIdentifier(ident);
+    boolean droppedFromCatalog =
         doWithCatalog(
-            getCatalogIdentifier(ident),
+            catalogIdent,
             c -> c.doWithSchemaOps(s -> s.dropSchema(ident, cascade)),
-            NonEmptySchemaException.class);
-
-    if (!dropped) {
-      return false;
-    }
+            NonEmptySchemaException.class,
+            RuntimeException.class);
 
+    // For unmanaged schema, it could happen that the schema:
+    // 1. Is not found in the catalog (dropped directly from underlying sources)
+    // 2. Is found in the catalog but not in the store (not managed by Gravitino)
+    // 3. Is found in the catalog and the store (managed by Gravitino)
+    // 4. Neither found in the catalog nor in the store.
+    // In all situations, we try to delete the schema from the store, but we don't take the
+    // return value of the store operation into account. We only take the return value of the
+    // catalog into account.
+    //
+    // For managed schema, we should take the return value of the store operation into account.
+    boolean droppedFromStore = false;
     try {
-      return store.delete(ident, SCHEMA, cascade);
+      droppedFromStore = store.delete(ident, SCHEMA, cascade);
+    } catch (NoSuchEntityException e) {
+      LOG.warn("The schema to be dropped does not exist in the store: {}", ident, e);
     } catch (Exception e) {
       throw new RuntimeException(e);
```

For unmanaged schema, should we throw an exception here?